dallinger/command_line/docker_ssh.py

```diff
@@ -367,36 +395,37 @@ def deploy(
     executor.run(
         f"docker compose -f ~/dallinger/{experiment_id}/docker-compose.yml run --rm web ls"
     )
-    print("Cleaning up db/user")
-    executor.run(
-        rf"""docker compose -f ~/dallinger/docker-compose.yml exec -T postgresql psql -U dallinger -c 'DROP DATABASE IF EXISTS "{experiment_id}";'"""
-    )
-    executor.run(
-        rf"""docker compose -f ~/dallinger/docker-compose.yml exec -T postgresql psql -U dallinger -c 'DROP USER IF EXISTS "{experiment_id}"; '"""
-    )
-    print(f"Creating database {experiment_id}")
-    executor.run(
-        rf"""docker compose -f ~/dallinger/docker-compose.yml exec -T postgresql psql -U dallinger -c 'CREATE DATABASE "{experiment_id}"'"""
-    )
-    create_user_script = f"""CREATE USER "{experiment_id}" with encrypted password '{postgresql_password}'"""
-    executor.run(
-        f"docker compose -f ~/dallinger/docker-compose.yml exec -T postgresql psql -U dallinger -c {quote(create_user_script)}"
-    )
     grant_roles_script = (
         f'grant all privileges on database "{experiment_id}" to "{experiment_id}"'
     )
+    if not update:
+        print("Cleaning up db/user")
+        executor.run(
+            rf"""docker compose -f ~/dallinger/docker-compose.yml exec -T postgresql psql -U dallinger -c 'DROP DATABASE IF EXISTS "{experiment_id}";'"""
+        )
+        executor.run(
+            rf"""docker compose -f ~/dallinger/docker-compose.yml exec -T postgresql psql -U dallinger -c 'DROP USER IF EXISTS "{experiment_id}"; '"""
+        )
+        print(f"Creating database {experiment_id}")
+        executor.run(
+            rf"""docker compose -f ~/dallinger/docker-compose.yml exec -T postgresql psql -U dallinger -c 'CREATE DATABASE "{experiment_id}"'"""
+        )
+        create_user_script = f"""CREATE USER "{experiment_id}" with encrypted password '{postgresql_password}'"""
```

Comment:
The fact that this line is skipped during `update` means that the new randomly generated `postgresql_password` that is uploaded to `docker-compose.yml` does not match the `postgresql_password` set in Postgres in the previous deployment. As a result I get a password authentication error when trying to run an updated experiment.

Reply:
Great! Works now